R/type_text.R

```diff
@@ -0,0 +1,44 @@
+#' Text annotations plot type
+#'
+#' @description Type function for adding text annotations to a plot. This function allows
+#' you to draw text at specified (x,y) coordinates. If no label is provided, the y-values 
+#' will be used as the text labels.
+#'
+#' @param labels Character vector of the same length as the number of x,y coordinates.
```

Comment:
`NA`s are already handled gracefully. I added a snapshot test to make sure things stay OK on that front.

There's a question about what we should do with `NA` labels: print the string "NA" or omit them entirely. I went with the latter, because that's the default behaviour of the `text()` function, and because it requires no special handling at all on our end.